pkg/operator/operator.go

```diff
@@ -404,8 +409,13 @@ func (o *Operator) cleanupOldResources(ctx context.Context) error {
 		return fmt.Errorf("get rule-evaluator Deployment: %w", err)
 	}
 	if _, ok := deploy.Annotations[o.opts.CleanupAnnotKey]; !ok {
-		if err := o.client.Delete(ctx, &deploy); client.IgnoreNotFound(err) != nil {
-			return fmt.Errorf("delete rule-evaluator Deployment: %w", err)
+		if err := o.client.Delete(ctx, &deploy); err != nil {
+			switch {
+			case apierrors.IsForbidden(err):
+				o.logger.Info("delete rule-evaluator Deployment was not allowed. Please remove it manually")
+			case !apierrors.IsNotFound(err):
+				return fmt.Errorf("delete rule-evaluator Deployment failed: %w", err)
```

Comment:
Question: Doesn't the Not Found error include the type (Deployment) and the name (rule-evaluator)? I think that might be redundant.

Reply:
This is checking for _not_ not found. However, as the error message generally includes the type and name, I updated the wording.